tensorboard/webapp/routes/feature_flag_serializer.ts

```diff
@@ -18,24 +18,23 @@ import {
   FeatureFlagMetadataMapType,
   FeatureFlagType,
 } from '../feature_flag/store/feature_flag_metadata';
-import {FeatureFlags} from '../feature_flag/types';
 
-export function featureFlagsToSerializableQueryParams<T extends FeatureFlags>(
-  overriddenFeatureFlags: Partial<FeatureFlags>,
+export function featureFlagsToSerializableQueryParams<T>(
+  overriddenFeatureFlags: Partial<T>,
   featureFlagMetadataMap: FeatureFlagMetadataMapType<T>
 ): SerializableQueryParams {
   return Object.entries(overriddenFeatureFlags)
     .map(([featureFlag, featureValue]) => {
       const featureFlagMetadata: FeatureFlagMetadata<any> =
-        featureFlagMetadataMap[featureFlag as keyof FeatureFlags];
+        featureFlagMetadataMap[featureFlag as keyof T];
       if (!featureFlagMetadata) {
         // No metadata for this feature flag. Shouldn't happen but we must
         // include the check for the compiler.
         // Return empty item. Will be filtered out.
         return {};
       }
       const key = featureFlagMetadata.queryParamOverride;
-      if (!key || featureValue === undefined) {
+      if (!key || (featureValue as FeatureFlagType) === undefined) {
```

Comment:
Out of curiosity, does the compiler actually complain about "featureValue === undefined" without the cast?

Reply:
No and I'm not sure how this got to be this way, last Friday is a bit of a haze